cvat-ui/src/components/annotation-page/top-bar/annotation-menu.tsx

```diff
@@ -52,6 +52,7 @@ function AnnotationMenuComponent(): JSX.Element {
 
     const finishJob = useCallback(() => {
         dispatch(finishCurrentJobAsync()).then(() => {
+            setJobState(JobState.COMPLETED);
```

Comment:
This does not fix the problem that is described in issue. When we click submit in single shape mode this code isnt even called.

Reply:
The root problem is that when `updateCurrentJobAsync` action updates the job, there is no changes in store, so the component doesnt know that job's state was changed and we need to re-render. As the reference to the jobInstance is not updated